File: python/BRIDGE_WHOLE_PART/BRIDGE_WHOLE_PART/whole_part/membresia/MembresiaVIP.py
from BRIDGE_WHOLE_PART.bridge.ImplAbstractaMembresia import ImplAbstractaMembresia
from BRIDGE_WHOLE_PART.whole_part.DatosBeneficios import DatosBeneficios

class MembresiaVIP(ImplAbstractaMembresia):
    
    datosBeneficios = DatosBeneficios()
    
    def __init__(self):
        pass
    
    def completarMembresia(self, datos):
        super().completarMembresia(datos[:-1])
        self.datosBeneficios.setDatosBeneficios(datos[-1])
    
    def mostrarMembresia(self):
        print(
        f'''
        Reservación:
        
        Datos cliente:
            Nombres          : {self.datosCliente.nombres}
            Apellido paterno : {self.datosCliente.apellidoPaterno}
            Apellido materno : {self.datosCliente.apellidoMaterno}
            Edad             : {self.datosCliente.edad}
            Tipo de cliente  : {self.datosCliente.tipoCliente}
            
        Beneficios   : {self.datosBeneficios.beneficios}
            
        '''
        )

File: php/BRIDGE_WHOLE_PART/whole_part/membresia/MembresiaVIP.php
<?php

require_once './bridge/ImplAbstractaMembresia.php';
require_once './whole_part/DatosBeneficios.php';
class MembresiaVIP extends ImplAbstractaMembresia{
    
    public $datosBeneficio;
    
    public function __construct() {
        parent::__construct();
    }
    
    public function completarMembreisa($datos) {
        parent::completarMembreisa($datos);
        $this->datosBeneficio = new DatosBeneficios();
        $this->datosBeneficio->setDatosBeneficios($datos["beneficios"]);
    }
    
    public function mostrarMembresia() {
        return parent::mostrarMembresia().$this->datosBeneficio->getDatosBeneficios(); 
    }
}
